akka-http-core/src/main/java/akka/http/javadsl/model/HttpHeader.java

```diff
@@ -44,4 +44,22 @@
      * Returns true if and only if the header is to be rendered in responses.
      */
     public abstract boolean renderInResponses();
+
+    /**
+     * Attempts to parse the given header name and value string into a header model instance.
+     *
+     * @throws IllegalArgumentException if parsing is unsuccessful.
+     */
+    public static HttpHeader parse(String name, String value) {
+      final akka.http.scaladsl.model.HttpHeader.ParsingResult result =
+        akka.http.scaladsl.model.HttpHeader.parse(name, value,
+          akka.http.impl.model.parser.HeaderParser$.MODULE$.DefaultSettings());
+
+      if (result instanceof akka.http.scaladsl.model.HttpHeader$ParsingResult$Ok) {
+        return ((akka.http.scaladsl.model.HttpHeader$ParsingResult$Ok) result).header();
+      }
+      else {
+        throw new IllegalArgumentException("Unable to parse header");
```

Comment:
In the best case, we would also support the ParsingResult in some way on the Java side. If we don't want to do that we could at least throw an `IllegalHeaderException` or `ParsingError` with the `ErrorInfo` in the error. Then the user can decide whether to expose only the `summary` of the error info or the full information.

Reply:
Alright. Switched to `IllegalHeaderException`

Reply:
Cool, thanks.